### PR TITLE
#5 Update DataCiteClient to be able to consume xml metadata; refactoring

### PR DESCRIPTION
**Pull request by mike-podolskiy90**

### CODE REVIEW

File: datacite-rest-client/src/it/java/org/gbif/datacite/rest/client/retrofit/DataCiteRetrofitClientSteps.java [hunk 1]
```diff
@@ -57,24 +62,32 @@ public void performGetDoiList() {
...
-        Datacite42Schema datacite42Schema = new Datacite42Schema();
-        datacite42Schema.setId(DOI_105);
-        datacite42Schema.setDoi(DOI_105);
+    public void performCreateDoi() throws IOException {
+        DoiSimplifiedModel model = new DoiSimplifiedModel();
+        model.setDoi(DOI_105);
+        model.setId(DOI_105);
+        String xmlMetadata = IOUtils.toString(
+                this.getClass().getResourceAsStream("/datacite-example-full-v4.xml"),
+                "UTF-8");
```
Comment (fmendezh) [inline datacite-rest-client/src/it/java/org/gbif/datacite/rest/client/retrofit/DataCiteRetrofitClientSteps.java]: there's constant for UTF-8 in java.nio.charset.StandardCharsets

File: datacite-rest-client/src/main/java/org/gbif/datacite/rest/client/model/DoiSimplifiedModel.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, doi, event, url);
+    }
+
+    @Override
+    public String toString() {
+        return "DoiSimplifiedModel{" +
```
Comment (fmendezh) [inline datacite-rest-client/src/main/java/org/gbif/datacite/rest/client/model/DoiSimplifiedModel.java]: we need to decide if we'll use SrtringJoiner instead, for example:
 @Override
    public String toString() {
        return new StringJoiner(", ", DoiSimplifiedModel.class.getSimpleName() + "[", "]")
                .add("id='" + id + "'")
                .add("doi='" + doi + "'")
                .add("event='" + event + "'")
                .add("xml='" + xml + "'")
                .add("url='" + url + "'")
                .toString();
    }

**Review by fmendezh — CHANGES_REQUESTED** [review]
a couple of tiny changes and suggestions

File: datacite-rest-client/src/it/java/org/gbif/datacite/rest/client/retrofit/DataCiteRetrofitClientSteps.java (hunk 1 above)
Comment (mike-podolskiy90) [reply datacite-rest-client/src/it/java/org/gbif/datacite/rest/client/retrofit/DataCiteRetrofitClientSteps.java]: done

File: datacite-rest-client/src/main/java/org/gbif/datacite/rest/client/model/DoiSimplifiedModel.java (hunk 2 above)
Comment (mike-podolskiy90) [reply datacite-rest-client/src/main/java/org/gbif/datacite/rest/client/model/DoiSimplifiedModel.java]: fixed

**Pull request merged**